rfc/model_evaluation/rfc.md

```diff
@@ -0,0 +1,49 @@
+|     model_uri              |   String        |   Required: False (Required only if mode is score or predict)  Description: MLFlow model uri (could be of type) ->        To Fetch models from an azureml run        - runs:/<azureml_run_id>/run-relative/path/to/model        To Fetch model from azureml model registry        - models:/<model_name>/<model_version>        - models:/<model_name>/<stage>  NOTE: model_uri has been added to accommodate current ML Designer as it doesn’t allow users to drag-and-drop Registered models    |
+|     mlflow_model           |   MLFlow Model  |   Required: False (Required only if mode is score or predict)  Description: MLFlow model (either registered or output of another job)                                                                                                                                                                                                                                                                                                                                                             |
+|     data_folder            |   URI Folder    |   Required: True  Description: Input folder which contains the following files ->        - Test Data (in csv format)        - Additional Parameters file (in JSON format)        - Any other artifact required for model evaluation like y_transformer (in pickled format)                                                                                                                                                                                                                        |
+|     test_data_file_name    |   String        |   Required: True  Description: Full path of test data file in above folder                                                                                                                                                                                                                                                                                                                                                                                                                        |
+|     additional_parameters  |   String        |   Required: False  Description: File name of the additional parameters JSON file.        Contents of JSON File:        - label_column_name: <Name of target column in test csv>                      (required if mode is score or compute_metrics)        - prediction_column_name: <Name of predictions column in test csv> (required if mode is compute_metrics)        - <Other parameters based on above task specific Metrics class from above SDK package>                                 |
```

Comment:
Thank you for elaborating, I still don't think this is a good design. We're basically avoiding having multiple specific components by having one single generic component with a big additional parameter field. I don't think this is promoting a better user experience.

If I compare two paths:
1. search for a specific component to evaluate my model/task and drag-n-drop in my graph with few changes
2. drag-n-drop the generic component then figure out the right additional parameters based on the docs

My opinion is that UX is better with path 1. I can be wrong, but I'd like this submitted to a test with users because I think the "generic component" design might lead to lots of pain points (finding the docs, figuring out the format, debugging, etc).

Reply:
Thanks for the very valid inputs Jeff. The 2 paths that you suggested indeed make sense. We have started engaging with UX team to run and gain User feedback on whether to have specific component for different tasks or keep a single Generic component. Based on the outcome, we plan to scope in the changes Post PrP.